acvm-repo/acvm/src/pwg/mod.rs

```diff
@@ -247,6 +262,13 @@ impl<'a, F: AcirField, B: BlackBoxFunctionSolver<F>> ACVM<'a, F, B> {
     }
 
     /// Finalize the ACVM execution, returning the resulting [`WitnessMap`].
+    pub fn finalize_with_profiling(self) -> (WitnessMap<F>, ProfilingSamples) {
```

Comment:
`finalize` could delegate to this for consistency in status handling.

Reply:
done